client/client.go

```diff
@@ -201,14 +202,20 @@ func (client *Client) WarningsSummary() (count int, timestamp time.Time) {
 	return client.warningCount, client.warningTimestamp
 }
 
+// RequestError is returned when there's an error creating an http.Request.
```

Comment:
This sounds too specific. We can return this for anything related to the request processing before the actual request is made. We don't need to mention the underlying implementation of performing the request here.